src/Util/Network.php

```diff
@@ -836,4 +836,29 @@ public static function unparseURL($parsed)
 			(strlen($query) ? "?".$query : '') .
 			(strlen($fragment) ? "#".$fragment : '');
 	}
+
+
+	/**
+	 * @brief Switch the scheme of an url between http and https
+	 *
+	 * @param string $url URL
+	 *
+	 * @return string switched URL
+	 */
+	public static function switchScheme($url)
+	{
+		$parts = parse_url($url);
```

Comment:
Please use the `PHP_URL_SCHEME` constant to limit the output of `parse_url()` to the only relevant bit.